src/compiler/checker.ts

```diff
@@ -15317,6 +15317,32 @@ namespace ts {
                 return caseType.flags & TypeFlags.Never ? defaultType : getUnionType([caseType, defaultType]);
             }
 
+            function getTypeFromName(type: Type, text: string) {

```

Comment:
Would prefer a more descriptive name like `getNarrowedTypeByTypeofResult`

Reply:
Went with `getImpliedTypeFromTypeofCase` to be consistent with terminology used in the documentation of `narrowBySwitchOnTypeOf`.